x-pack/plugins/maps/public/connected_components/layer_addpanel/index.js

```diff
@@ -34,12 +34,12 @@ function mapStateToProps(state = {}) {
 
 function mapDispatchToProps(dispatch) {
   return {
-    viewLayer: async layer => {
+    previewLayer: async layerDescriptor => {
       await dispatch(setSelectedLayer(null));
       await dispatch(removeTransientLayer());
-      dispatch(addLayer(layer.toLayerDescriptor()));
```

Comment:
`ILayer#toLayerDescriptor` is now stale code.

Can remove:
- https://github.com/elastic/kibana/blob/939bf2a6ec45154d33e982550075417d91fe9eeb/x-pack/plugins/maps/public/layers/layer.tsx#L65
- https://github.com/elastic/kibana/blob/939bf2a6ec45154d33e982550075417d91fe9eeb/x-pack/plugins/maps/public/layers/layer.tsx#L416-L418